src/dbnode/encoding/proto/iterator.go

```diff
@@ -185,6 +185,14 @@ func (it *iterator) Next() bool {
 				it.err = fmt.Errorf("%s error reading custom fields schema: %v", itErrPrefix, err)
 				return false
 			}
+
+			// When the encoder changes its schema it will reset all of its nonCustomFields state
+			// which means that the iterator needs to do the same to keep them synchronized at
```

Comment:
@richardartoul  you mentioned there is some bug around schema change, is this it? 
I saw you made some changes for encoder.SetSchema but later reverted it.

Which test caught the bug?

Reply:
@Haijuncao `TestBufferSnapshot` caught it. I added a call to `SetSchema` in `RoundTripPropTest` to reproduce the bug and prevent a regression.

I reverted my change because I overcomplicated it last night. This fix is simpler than what I did last night